homeassistant/components/wittiot/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""The Wittiot integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+
+from aiohttp import ClientSession
+from aiohttp.client_exceptions import ClientConnectorError
+from wittiot import API
+from wittiot.errors import WittiotError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_IP, CONNECTION_TYPE, DOMAIN, LOCAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Wittiot from a config entry."""
+
+    websession = async_get_clientsession(hass)
+    update_interval = timedelta(seconds=60)
+    if not entry.unique_id:
+        hass.config_entries.async_update_entry(entry, unique_id=entry.data[CONF_IP])
```

Comment:
This will be a core integration, so if we start marking entries with unique_ids we dont have to guard for this anymore